pkg/runtime/reconciler.go

```diff
@@ -1045,6 +1055,35 @@ func (r *resourceReconciler) getRegion(
 	return ackv1alpha1.AWSRegion(r.cfg.Region)
 }
 
+// getDeletionPolicy returns the resource's deletion policy based on the default
+// behaviour or any other overriding annotations.
+//
+// We look for the deletion policy in the annotations based on the following
+// precedence:
+//   - The resource's `services.k8s.aws/deletion-policy` annotation, if present
+//   - The resource's Namespace's `{service}.services.k8s.aws/deletion-policy` annotation, if present
+//   - The controller's `--deletion-policy` CLI flag
+func (r *resourceReconciler) getDeletionPolicy(
+	res acktypes.AWSResource,
+) ackv1alpha1.DeletionPolicy {
+	// look for deletion policy in CR metadata annotations
+	resAnnotations := res.MetaObject().GetAnnotations()
+	deletionPolicy, ok := resAnnotations[ackv1alpha1.AnnotationDeletionPolicy]
```

Comment:
I also didn't know it is possible to use type aliases for map access..

Reply:
`AnnotationDeletionPolicy` is a string - nothing weird going on here.